src/archivematicaCommon/lib/databaseFunctions.py

```diff
@@ -20,84 +20,181 @@
+def insertIntoEvents(fileUUID="", eventIdentifierUUID="", eventType="", eventDateTime=getUTCDate(), eventDetail="", eventOutcome="", eventOutcomeDetailNote=""):
+    """
+    Creates a new entry in the Events table using the supplied arguments.
+
+    :param str fileUUID: The UUID of the file with which this event is associated. Can be blank.
+    :param str eventIdentifierUUID: The UUID for the event being generated. If not provided, a new UUID will be calculated using the version 4 scheme.
+    :param str eventType: Can be blank.
+    :param datetime.datetime eventDateTime: The time at which the event occurred. If not provided, the current date will be used.
+    :param str eventDetail: Can be blank. Will be used in the eventDetail element in the AIP METS.
+    :param str eventOutcome: Can be blank. Will be used in the eventOutcome element in the AIP METS.
+    :param str eventOutcomeDetailNote: Can be blank. Will be used in the eventOutcomeDetailNote element in the AIP METS.
+    """
     agent = getAgentForFileUUID(fileUUID)
-    if eventIdentifierUUID == "":
-        eventIdentifierUUID = uuid.uuid4().__str__()
-    databaseInterface.runSQL("""INSERT INTO Events (fileUUID, eventIdentifierUUID, eventType, eventDateTime, eventDetail, eventOutcome, eventOutcomeDetailNote, linkingAgentIdentifier)
-        VALUES (%s, %s, %s, %s, %s, %s, %s, %s)""",
-        (fileUUID, eventIdentifierUUID, eventType, eventDateTime,
-         eventDetail, eventOutcome, eventOutcomeDetailNote, agent)
-    )
+    if not eventIdentifierUUID:
+        eventIdentifierUUID = (uuid.uuid4())
+
+    Event.objects.create(event_id=eventIdentifierUUID, file_uuid_id=fileUUID,
+                         event_type=eventType, event_datetime=eventDateTime,
+                         event_detail=eventDetail, event_outcome=eventOutcome,
+                         event_outcome_detail=eventOutcomeDetailNote,
+                         linking_agent_id=agent)
 
 def insertIntoDerivations(sourceFileUUID="", derivedFileUUID="", relatedEventUUID=""):
-    databaseInterface.runSQL("""INSERT INTO Derivations
-        (sourceFileUUID, derivedFileUUID, relatedEventUUID)
-        VALUES (%s, %s, %s)""",
-        (sourceFileUUID, derivedFileUUID, relatedEventUUID)
-    )
+    """
+    Creates a new entry in the Derivations table using the supplied arguments. The two files in this relationship should already exist in the Files table.
+
+    :param str sourceFileUUID: The UUID of the original file.
+    :param str derivedFileUUID: The UUID of the derived file.
+    :param str relatedEventUUID: The UUID for an event describing the creation of the derived file. Can be blank.
+    """
+    if not sourceFileUUID:
+        raise ValueError("sourceFileUUID must be specified")
+    if not derivedFileUUID:
+        raise ValueError("derivedFileUUID must be specified")
```

Comment:
All python functions can be called by specifying their args in order, or by naming them regardless of how the parameter is defined (http://www.diveintopython.net/power_of_introspection/optional_arguments.html) so it shouldn't affect the places this is called.

Probably worthwhile to keep the checks to catch things like ```""``` since those errors are almost certainly better than what Django generates if the constraint fails.